### PR TITLE
Support ingestion of inline, data-uri sourcemaps in input .css

### PR DESCRIPTION
**Pull request by cspotcode**
This patch enables clean-css to ingest sourcemaps within input .css files where the sourcemap is an inline data URI.

### CODE REVIEW

**Comment by jakubpawlowicz** [comment]
@cspotcode Thanks for the PR - I'll adapt your code a bit and merge it manually.

It would be nice to have a testcase too - it can be sth as simple as https://github.com/jakubpawlowicz/clean-css/blob/master/test/source-map-test.js#L428 with source as Base64 https://github.com/jakubpawlowicz/clean-css/blob/master/test/fixtures/source-maps/styles.css.map

File: lib/utils/input-source-map-tracker.js [hunk 1]
```diff
@@ -63,15 +66,26 @@ function fromSource(self, data, whenDone, context) {
...
-        var sourceDir = sourceFile ? path.dirname(sourceFile) : self.options.relativeTo;
-        var sourceMapPath = path.resolve(self.options.root, path.join(sourceDir || '', sourceMapFile));
-
-        var sourceMapData = fs.readFileSync(sourceMapPath, 'utf-8');
-        self.trackLoaded(sourceFile || undefined, sourceMapPath, sourceMapData);
+        var sourceMapPath, sourceMapData;
+        if(isDataUri) {
+          // source map's path is the same as the source file it comes from
+          sourceMapPath = path.resolve(self.options.root, sourceFile);
+          var buffer = dataUriToBuffer(sourceMapFile);
```
Comment (jakubpawlowicz) [inline lib/utils/input-source-map-tracker.js]: Can't we just use `new Buffer(sourceMapFile, 'base64').toString()`?

Comment (cspotcode) [inline lib/utils/input-source-map-tracker.js]: You need the data URI parser to strip off the data URI's mimetype and grab
its charset declaration.  Data URIs, surprisingly, default to ASCII
encoding, so it's important to respect the data URI's charset when parsing
it.  See also: https://github.com/stylus/stylus/issues/1935

Comment (jakubpawlowicz) [inline lib/utils/input-source-map-tracker.js]: I'm pretty sure we can do without those 2 dependencies given what they do: https://github.com/TooTallNate/node-data-uri-to-buffer/blob/master/index.js#L44

Comment (cspotcode) [inline lib/utils/input-source-map-tracker.js]: It's your decision, depending on your philosophy about dependencies.  Given
that they're small, simple libraries, I don't see a practical downside to
using them.

You'll also need to handle data URIs that are not base64 encoded.

Comment (jakubpawlowicz) [inline lib/utils/input-source-map-tracker.js]: The idea is to keep dependencies to the minimum. I made some small changes to your code and pushed it as #604 - let me know what you think.

**Comment by jakubpawlowicz** [comment]
Superseded by #604.

**Pull request closed without merging**